### ISSUE TITLE
Editor unusable when class has many Sonar Issues at class level 

### ISSUE THREAD

**Issue by tpischke** [sonar-intellij-plugin/sonar-intellij-plugin#110]
In Java classes that have many issues at the class level (code coverage, package cycles, etc), the issues are displayed at the top of the editor window.  Unfortunately, they cannot be scrolled out of view, so they take up valuable screen space.  Worse, in very bad classes with many such warnings, it becomes entirely impossible to edit the file at all.  

This makes the plugin unusable for us, since certain critical classes cannot be edited without disabling the inspections.  An acceptable fix would be to make it possible to scroll the issues out of view.  Another acceptable fix would be to make their display somehow optional, via folding or similar.

Please don't suggest that we simply fix the code issues, since we are dealing with legacy code where that simply isn't a short-term option. 

Please see the attached screenshot for an example of an extreme case where the editor becomes unusable due to many such issues.

![sonarqubeplugin_issue](https://cloud.githubusercontent.com/assets/2378635/5917559/2ddd70ac-a61e-11e4-8aef-67da11608a3b.png)

**Comment by omayevskiy** [comment]
Fix the code :dancer: 

this happens if an inspection does not have any line of code, then it is shown on file level. Another option would be to show them on first line.

**Comment by tpischke** [comment]
That's true, but i fear that it may take a long time before Intellij would fix the issue when we report it there.  Could the plugin display them on the first line as a workaround until intellij is patched?  I think this plugin has a much faster release turnaround and bug fix rate than intellij;)

**Comment by tpischke** [comment]
https://youtrack.jetbrains.com/issue/IDEA-135658

**Comment by omayevskiy** [comment]
thanks for the flowers ;-)
I would recommend you to describe the issue in more details at jetbrains and not just linking it to here.

**Comment by tpischke** [comment]
Added more detail in the IntelliJ Issue :)

**Comment by omayevskiy** [comment]
from jetbrains discussion:

It would make sense to have file level annotation saying something like "Rule foo bar violated (100 times)"
And after clicking or mouse over it get the full list with detailed messages.

**Comment by tpischke** [comment]
Any further thoughts on this?  It's still a blocker for us.  Your suggestion from from 5 days ago would probably ease our pain considerably until Intellij fixes the issue on their side.

**Comment by omayevskiy** [comment]
thoughts no, it "just" needs to be done ;-)
in the mean time you could deactivate that blocker rule
